docs/source/concepts/ledger-model/ledger-structure.rst

```diff
@@ -185,6 +189,12 @@ below shows all its proper subtransactions on the right (yellow boxes).
    :align: center
    :width: 100%
 
+To illustrate contract keys, suppose that the contract key for a `PaintOffer` consists of the reference number and the painter as the key maintainer.
+So Alice can refer to the `PaintOffer` by its key `(P, P123)`.
+To make this explicit, we use the notation `PaintOffer @P A &P123` for contracts, where `@` marks the maintainers and `&` the other key parts.
+The ledger integrity constraints in the next section ensure that there is always at most one active `PaintOffer` for a given key.
+So if the painter retracts its `PaintOffer` and later Alice tries to accept it, she can then record the absence with a `NoKey (P, P123)` key assertion.
```

Comment:
Can she? I would have thought she's not able to, as she's not the maintainer.

Reply:
The restriction that the submitter is a maintainer is only a limitation of the currently available implementations. It's nothing that we want to enforce on the abstract ledger model. For if we eventually figure out how to do disclosure properly, then the restriction will go away.